### ISSUE TITLE
Gem has permission issues

### ISSUE THREAD

**Issue by benmullin333** [signalfx/signalfx-ruby#23]
Hi!

Thanks for taking care of #20 so quickly!

I believe your permissions are borked in the gem.  
Several files are set to `640` when they should be set to `644`.
This is blocking the ability for multiple users to share one ruby installation on the same system.

Here's what you have:
```[bmullin@flybox-007 signalfx-2.0.1]# pwd
/usr/local/rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/signalfx-2.0.1

[bmullin@flybox-007 signalfx-2.0.1]# ls -al
total 60
drwxr-xr-x   5 bmullin bmullin  4096 Jan 17 19:49 .
drwxr-xr-x 189 bmullin bmullin 12288 Jan 17 21:45 ..
drwxr-xr-x   2 bmullin bmullin  4096 Jan 17 19:49 bin
drwxr-xr-x   2 bmullin bmullin  4096 Jan 17 19:49 examples
-rw-r-----   1 bmullin bmullin   177 Jan 17 19:49 Gemfile
-rw-r-----   1 bmullin bmullin  2613 Jan 17 19:49 Gemfile.lock
-rw-r-----   1 bmullin bmullin   695 Jan 17 19:49 .gitignore
drwxr-xr-x   4 bmullin bmullin  4096 Jan 17 19:49 lib
-rw-r-----   1 bmullin bmullin   117 Jan 17 19:49 Rakefile
-rw-r-----   1 bmullin bmullin  5554 Jan 17 19:49 README.md
-rw-r-----   1 bmullin bmullin  2032 Jan 17 19:49 signalfx.gemspec
-rw-r-----   1 bmullin bmullin    59 Jan 17 19:49 .travis.yml
```

Here's an example of what I think works better.
```
[bmullin@flybox-007 signalfx-2.0.1]# ls -al ../thor-0.19.1/
total 56
drwxr-xr-x   5 bmullin bmullin  4096 Oct 24 21:49 .
drwxr-xr-x 189 bmullin bmullin 12288 Jan 17 21:45 ..
drwxr-xr-x   2 bmullin bmullin  4096 Oct 24 21:49 bin
-rw-r--r--   1 bmullin bmullin  5946 Oct 24 21:49 CHANGELOG.md
-rw-r--r--   1 bmullin bmullin    49 Oct 24 21:49 .document
drwxr-xr-x   3 bmullin bmullin  4096 Oct 24 21:49 lib
-rw-r--r--   1 bmullin bmullin  1075 Oct 24 21:49 LICENSE.md
-rw-r--r--   1 bmullin bmullin  1462 Oct 24 21:49 README.md
drwxr-xr-x   9 bmullin bmullin  4096 Oct 24 21:49 spec
-rw-r--r--   1 bmullin bmullin   692 Oct 24 21:49 Thorfile
-rw-r--r--   1 bmullin bmullin   916 Oct 24 21:49 thor.gemspec
```

Thanks!
Ben

**Comment by keitwb** [comment]
@benmullin333 Are you sure you don't have some kind of umask or gem config issue going on with your local copy?  According to the latest commit on the master branch in git, the file permissions are 0644 (rw-r--r--) for all of those files that show as 0640 for you:

```sh
$ git ls-tree -l HEAD
100644 blob 6bb793f815eda6b3670448ee7b0928e9ef410651     695    .gitignore
100644 blob 7fa6097dc17f94c143a33d621760c553ac5b3803      59    .travis.yml
100644 blob ce9e256cfba400596f3fd02e690c61bb33545483     177    Gemfile
100644 blob 366b3c8c18e091640fcc28f954b27a44295c96b2    2613    Gemfile.lock
100644 blob 0686896c79c5c7172b9a7ae57824b0ebac9c3fa8    5635    README.md
100644 blob b7e9ed549be37412bd2f7e90f0b33165e3052ceb     117    Rakefile
040000 tree d1bd5642a21234b6254037d0f77d60fa5f458a32       -    bin
040000 tree 3100cf9418cce66e622b79a96c3aaa64bfdbafc1       -    examples
040000 tree e5c488e074b0c0134b102fec7376e136c4640746       -    lib
100644 blob 4c83cfb87e7d298a9a7f211beebb312a4a09b022    2032    signalfx.gemspec
040000 tree 39bad60c7a9c281e2fe987800d1ab6ae156b4e0b       -    spec
```

It shows the same thing in the Github UI when you click on _Blame_ for a file.  I am by no means an expert on how git handles file permissions so I could be overlooking something here, but it seems like the repo itself has the right permissions.  I have no idea what is going on.

**Comment by benmullin333** [comment]
hmm.. maybe this is related to the machine that built the gem?

This page leads me to think that a umask on a machine building the gem could cause it.
https://tickets.opscode.com/browse/KNIFE-77

**Comment by benmullin333** [comment]
I don't think we have any issues with umask on our end.  we've been able to replicate this on our laptops and in our aws linux nodes.  We're not aware of any special umask changes.

**Comment by edgeworthch** [comment]
Looks like the gemfile on rubygems.org is where the `chmod o-rwx` happened.  When I unarchive https://rubygems.org/downloads/signalfx-2.0.1.gem I see the same file permissions that @benmullin333 reports above.

Do y'all control the rubygems.org gemfile, or should we follow up with someone else?

**Comment by keitwb** [comment]
Yes, I was able to reproduce the problem by using the gem from rubygems.org.  I repackaged the gem for the 2.0.2 release (which includes the fix in #20) and it appears to have the right permissions locally so I'l release that later today once I get our rubygems.org password from the right person and confirm that the permissions are right.  I'll close this ticket when it is right.

Thanks for the help debugging this.
